utilities/pytest_utils.py

```
@@ -52,6 +55,72 @@
+    recorded: bool = False
+
+
+_failure_info = _FailureInfo()
+
+
+class InjectFailurePlugin:
+    """Plugin to inject synthetic failure testcase into JUnit XML before LogXML writes."""
+
+    @staticmethod
+    @pytest.hookimpl(tryfirst=True)
+    def pytest_sessionfinish(session: pytest.Session, exitstatus: int) -> None:
```

Ruff (0.15.21)

[warning] Unused static method argument: `exitstatus`

(ARG004)